infrastructure/lambda/redirect/index.mjs

```diff
@@ -0,0 +1,22 @@
+// @ts-check
+export const handler = async (event) => {
```

Comment:
Enjoyed comparing this to the Haskell code you sent, I'm in over my head with Haskell syntax but it helped me understand some of it better!